### ISSUE TITLE
logic bug in cache_nvidia_string_value_update()

### ISSUE THREAD

**Issue by z1atk0** [brndnmtthws/conky#1178]
**Issue**

While hunting down #1177 in `src/nvida.cc`, I found another logic bug in the [`cache_nvidia_string_value_update()`](https://github.com/brndnmtthws/conky/blob/main/src/nvidia.cc#L810) function. It doesn't do any actual damage, but it prevents `${nvidia perfmin}` and `${nvidia perfmax}` from being cached properly:
```
  } else if (strcmp(token, (char *)"perf") == 0 &&
             ac_string[gid].memTransferRatemax < 0) {
    if (search == SEARCH_MIN) {
      ac_string[gid].perfmin = *value;
    } else if (search == SEARCH_MAX) {
      ac_string[gid].perfmax = *value;
    }
  }
```
It doesn't make any sense to check if `memTransferRatemax` is already cached, when trying to decide if we should cache either `perfmin` or `perfmax`. As both values share a single token (`"perf"`), the is-it-already-cached?-check needs to go along with the respective search type check. This could look like this, for example:
```
  } else if (strcmp(token, (char *)"perf") == 0) {
    if (search == SEARCH_MIN &&
        ac_string[gid].perfmin < 0) {
      ac_string[gid].perfmin = *value;
    } else if (search == SEARCH_MAX &&
               ac_string[gid].perfmax < 0) {
      ac_string[gid].perfmax = *value;
    }
  }
```

**Information**

`conky-1.12.2`

**Comment by z1atk0** [comment]
Still present in `conky-1.18.1`.

**Comment by z1atk0** [comment]
Still present in `conky-1.19.8`.

**Comment by brndnmtthws** [comment]
Would you like to open a PR?

**Comment by w0j0pl** [comment]
> Would you like to open a PR?
I made another commit in the same fork and new pull request with suggested change from this issue.
Please let me know if that's ok, or I should make another fork next time, because idk what is a good practice.

**Comment by brndnmtthws** [comment]
Should be fixed with #2022.
